src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -982,48 +982,27 @@ _mongoc_client_recv (mongoc_client_t *client,
 }
 
 
-/*
- *--------------------------------------------------------------------------
- *
- * mongoc_client_new --
- *
- *       Create a new mongoc_client_t using the URI provided.
- *
- *       @uri should be a MongoDB URI string such as "mongodb://localhost/"
- *       More information on the format can be found at
- *       http://docs.mongodb.org/manual/reference/connection-string/
- *
- * Returns:
- *       A newly allocated mongoc_client_t or NULL if @uri_string is
- *       invalid.
- *
- * Side effects:
- *       None.
- *
- *--------------------------------------------------------------------------
- */
 mongoc_client_t *
 mongoc_client_new (const char *uri_string)
 {
-   mongoc_topology_t *topology;
    mongoc_client_t *client;
    mongoc_uri_t *uri;
-
+   bson_error_t error = {0};
 
    if (!uri_string) {
       uri_string = "mongodb://127.0.0.1/";
    }
 
-   if (!(uri = mongoc_uri_new (uri_string))) {
+   if (!(uri = mongoc_uri_new_with_error (uri_string, &error))) {
+      /* Log URI errors as a warning for consistency with mongoc_uri_new */
+      MONGOC_WARNING ("Error parsing URI: '%s'", error.message);
```

Comment:
Per the [Connection String spec](https://github.com/mongodb/specifications/blob/master/source/connection-string/connection-string-spec.rst#values), URI errors should be logged as warnings. This change keeps `mongoc_client_new` consistent with previous versions of libmongoc, where the function only relied on behavior within `mongoc_uri_new`. Additionally, this fixes the "/Client/max_staleness" test failure @kevinAlbs noted in https://github.com/mongodb/mongo-c-driver/pull/921#discussion_r775611909

Reply:
Good catch, I was not aware the Connection String spec was noted the log level.